pkg/system/installation.go

```diff
@@ -17,35 +19,157 @@ import (
 
 const DBXRootSecret = "yes-i-will-destroy-everything-on-this-disk"
 
-func GetInstallationMode(dbxState dogeboxd.DogeboxState) (dogeboxd.BootstrapInstallationMode, error) {
-	// First, check if we're already installed.
-	if _, err := os.Stat("/opt/dbx-installed"); err == nil {
-		return dogeboxd.BootstrapInstallationModeIsInstalled, nil
-	} else if !os.IsNotExist(err) {
-		return "", fmt.Errorf("error checking for RO installation media: %v", err)
+const (
+	one_gigabyte            = 1024 * 1024 * 1024
+	ten_gigabytes           = 10 * one_gigabyte
+	three_hundred_gigabytes = 300 * one_gigabyte
+	installationMediaPath   = "/opt/ro-media"
+)
+
+func logToWebSocket(t dogeboxd.Dogeboxd, message string) {
+	log.Printf("logging to web socket: %s", message)
+	t.Changes <- dogeboxd.Change{
+		ID:     "recovery",
+		Type:   "recovery",
+		Update: message,
 	}
+}
+
+func IsInstalled(t dogeboxd.Dogeboxd, dbxState dogeboxd.DogeboxState) (bool, error) {
+	logToWebSocket(t, "checking if Dogebox OS is already installed")
+	return checkNixOSDisksForFile(t, "/opt/dbx-installed")
+}
 
-	// If we're not already installed, but we've been configured, no install for you.
+func GetInstallationMode(t dogeboxd.Dogeboxd, dbxState dogeboxd.DogeboxState) (dogeboxd.BootstrapInstallationMode, error) {
+	// If we've been configured, no install for you.
 	if dbxState.InitialState.HasFullyConfigured {
 		return dogeboxd.BootstrapInstallationModeCannotInstall, nil
 	}
 
 	// Check if we're running on RO installation media. If so, must install.
-	if _, err := os.Stat("/opt/ro-media"); err == nil {
-		return dogeboxd.BootstrapInstallationModeMustInstall, nil
-	} else if !os.IsNotExist(err) {
+	isInstallationMedia, err := isInstallationMedia(t, "")
+	if err != nil {
 		return "", fmt.Errorf("error checking for RO installation media: %v", err)
 	}
+	if isInstallationMedia {
+		return dogeboxd.BootstrapInstallationModeMustInstall, nil
+	}
 
 	// Otherwise, the user can optionally install.
-	return dogeboxd.BootstrapInstallationModeCanInstalled, nil
+	return dogeboxd.BootstrapInstallationModeCanInstall, nil
 }
 
-const (
-	one_gigabyte            = 1024 * 1024 * 1024
-	ten_gigabytes           = 10 * one_gigabyte
-	three_hundred_gigabytes = 300 * one_gigabyte
-)
+func isInstallationMedia(t dogeboxd.Dogeboxd, mountPoint string) (bool, error) {
```

Comment:
This function can be rewritten to be a little more go-idiomatic:

```suggestion
func isInstallationMedia(t dogeboxd.Dogeboxd, mountPoint string) (bool, error) {
	roMediaPath := filepath.Join(mountPoint, installationMediaPath)

	var isMedia bool
	if _, err := os.Stat(roMediaPath); err != nil {
		if !os.IsNotExist(err) {
			logToWebSocket(t, fmt.Sprintf("error checking installation media flag: %v", err))
			return false, err
		}
		isMedia = false
	} else {
		isMedia = true
	}

	logToWebSocket(t, fmt.Sprintf("mount point %s is installation media? %v", mountPoint, isMedia))
	return isMedia, nil
}
```

Reply:
Agreed